lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -432,20 +441,71 @@ void LowerClassesPass::lowerClassExtern(ClassExternOp classExternOp,
 }
 
 // Update Object instantiations in a FIRRTL Module or OM Class.
-void LowerClassesPass::updateInstances(Operation *op) {
+LogicalResult LowerClassesPass::updateInstances(Operation *op) {
   OpBuilder builder(op);
   // For each Object instance.
   for (auto firrtlObject : llvm::make_early_inc_range(
            op->getRegion(0).getOps<firrtl::ObjectOp>())) {
+
+    // build a table mapping the indices of input ports to their position in the
+    // om class's parameter list.
```

Comment:
Is it safe to assume the order of the input ports matches the order in the om class parameter list? I think it should be deterministic, and I was trying to rely on that in my PR. If not, I guess it's safer to map explicitly, and I should use this table in my PR as well for handling instance ops.

Reply:
If we were iterating the results of a firrtl.instance op, it would be okay. For objects, it's different:

Here is an example:
```mlir
firrtl.circuit "Top" {
  firrtl.module @Top() {}

  firrtl.class @TwoInputs(in %a: !firrtl.integer, in %b: !firrtl.integer) { }

  firrtl.class @AssignInputsOutOfOrder() {
    %x = firrtl.integer 123
    %y = firrtl.integer 456
    %obj = firrtl.object @TwoInputs(in a: !firrtl.integer, in b: !firrtl.integer)

    %obj_a = firrtl.object.subfield %obj[a] : !firrtl.class<@TwoInputs(in a: !firrtl.integer, in b: !firrtl.integer)>
    firrtl.propassign %obj_a, %x : !firrtl.integer

    %obj_b = firrtl.object.subfield %obj[b] : !firrtl.class<@TwoInputs(in a: !firrtl.integer, in b: !firrtl.integer)>
    firrtl.propassign %obj_b, %y : !firrtl.integer
  }
}
```

`subfield %obj[b]` is deserialized last, so it will be at the front of the use-list for `%obj`, and so it's driver becomes the first arg to the om object:
```
    %0 = "firrtl.integer"() {value = 123 : si12} : () -> !firrtl.integer
    %1 = "firrtl.integer"() {value = 456 : si12} : () -> !firrtl.integer
    %2 = "om.object"(%1, %0) {className = "TwoInputs"} : (!firrtl.integer, !firrtl.integer) -> !om.class.type<@TwoInputs>
```

Reply:
Ahh ok, I was coming from the angle of iterating results of firrtl.instance, now it makes sense.